client/src/pages/record.tsx

```diff
@@ -167,6 +214,65 @@ function RecordPage(props: {
   const warnEmptyTranscript =
     curAnswer?.attentionNeeded === AnswerAttentionNeeded.NEEDS_TRANSCRIPT;
 
+  const [transcriptText, setTranscriptText] = useState<string>("");
+  const [editorState, setEditorState] = useState<EditorState>(
+    EditorState.createWithContent(ContentState.createFromText(""))
+  );
+  const markdownConfig = {
+    blockTypesMapping: {
+      /* mappings */
+    },
+    emptyLineBeforeBlock: true,
+  };
+
+  // Used to get the initial plain text
+  function getTranscriptText(text: string) {
+    setTranscriptText(text);
+    const _editorState = EditorState.createWithContent(
+      ContentState.createFromText(text)
+    );
+    setEditorState(_editorState);
+    return transcriptText;
+  }
+
+  // Get markdown from editor state
+  function getMarkdown(contentState: ContentState) {
+    const markdown = stateToMarkdown(
+      contentState, //editorState.getCurrentContent()
+      markdownConfig
+    );
+    return markdown;
+  }
+
+  // Get draftjs raw content from editor state
+  function getDraftContent(contentState: ContentState) {
+    const draftContent = convertToRaw(editorState.getCurrentContent());
+    return draftContent;
+  }
+
+  // Adaptor function to convert draftjs to markdown
+  function draftToMarkdown(contentState: ContentState) {
```

Comment:
this function seems redundant since it just returns another function with the same params?

Reply:
Yeah this and markdownToDraft both return the same things as those other two functions so I'm removing them.